Add artifact locations to RPM README

CloudNiner opened the pull request:
Also adds some note about some of the caveats a user has to navigate when dealing with NetCDF and S3 credentials on certain EC2 instance types

CloudNiner committed:
Add artifact locations to RPM README

CloudNiner committed:
Merge branch 'master' into feature/awf/rpm-readme-tweaks

CloudNiner commented:
@jpolchlo updated this README after merging #1. This is ready for a look in order to be merged.

jpolchlo commented on rpms/README.md:
```diff
@@ -5,53 +5,74 @@ needed to build a set of RPMs that are useful for working with Geotrellis.  If
 you are not intending to alter the composition or versions of the RPM
 packages, then you should be able to use the published binaries described in
 the README at the root of this project and do not need to build these
```
There is no description of pre-built RPMs in the root README.  This sentence should be fixed.

CloudNiner replied:
Done - it looks like that was removed at some point.

jpolchlo commented on rpms/README.md:
````diff
@@ -5,53 +5,74 @@ needed to build a set of RPMs that are useful for working with Geotrellis.  If
-  > **Fetch, do not build** if you have access to an already-built set of
-  > RPMs, and your purpose is to use these RPMs to build a Docker container.
+```
+s3://geotrellis-build-artifacts/rpms/
````
:+1:

jpolchlo commented on rpms/README.md:
```diff
@@ -5,53 +5,74 @@ needed to build a set of RPMs that are useful for working with Geotrellis.  If
+Built RPMs are publicly available at:
 
-  > **Fetch, do not build** if you have access to an already-built set of
-  > RPMs, and your purpose is to use these RPMs to build a Docker container.
```
I would like to see a quick explanation at the top of this README as to when a user needs to to what.  Actually building these RPMs should almost never happen.  It should be 100% crystal clear, for instance, that if the intention is to build a docker image, the RPMs need to be at least fetched first.  Removing these blocks I think makes it less easy to figure out how to use the repo.  Unless they are moved to the READMEs under the `emr` and `docker` subdirectories.

CloudNiner replied:
This was a terrible merge job, I'm never using the GitHub UI again. Updated to re-add the list of options that the user is presented with, and added a new "build and publish" option as well, along with links to the relevant section in this document.

jpolchlo commented on rpms/README.md:
```diff
@@ -5,53 +5,74 @@ needed to build a set of RPMs that are useful for working with Geotrellis.  If
 
-If you have a special use case, and the pre-built binaries are not sufficient,
-then you may need to build them yourself.
+If you need to load them during an EMR bootstrap, you can use `s3://geotrellis-build-artifacts/rpms/bootstrap.sh`.
```
The standard approach presented by this repo is to use the Makefiles and configs in the `emr` directory.  Presenting the bootstrap script alone may cause confusion.

We also have no automatic means to ensure that this bootstrap script on S3 stays up to date, so that should be addressed if it's somehow not sufficient to supply the bootstrap in `emr`.

CloudNiner replied:
> The standard approach presented by this repo is to use the Makefiles and configs in the emr directory

True. I removed the ref to bootstrap.sh here in order to keep a consistent story.

> We also have no automatic means to ensure that this bootstrap script on S3 stays up to date

Also true. That's outside the scope of this PR, I created a new issue: https://github.com/geotrellis/geotrellis-deployments/issues/6

jpolchlo commented on rpms/README.md:
```diff
@@ -5,53 +5,74 @@ needed to build a set of RPMs that are useful for working with Geotrellis.  If
+  "Install GDAL 2.4 dependencies",
+  "s3://geotrellis-build-artifacts/rpms/bootstrap.sh",
+  "s3://geotrellis-build-artifacts", "gdal-2.4.1"
+)
```
This block is provided without context, and it's not clear what this is or how to use it.

CloudNiner replied:
I moved this to a new subsection at the end titled "working with SBT lighter" and referenced it in the first section. We've used both the makefiles and sbt lighter on various projects, figured it was worth documenting that usage here.

jpolchlo commented on rpms/README.md:
```diff
@@ -5,53 +5,74 @@ needed to build a set of RPMs that are useful for working with Geotrellis.  If
 `s3://bucket/prefix/` is the path to a "directory" on S3 where RPMs have been
 previously published.  This will download the available RPMs as if you had
 built them yourself.  You may then go to the `docker` directory off the
 project root and follow the instructions for building the image.
```
In the interest of the crystal clarity mentioned in previous comment, maybe we shouldn't bury this all the way down here (unless we're providing some redundancy for documentation skimmers).

CloudNiner replied:
I believe this is at least partially addressed by adding a document relative link to the first section for those that are interested in fetching. Think it should still be better highlighted after the latest changes?

jpolchlo commented on rpms/README.md:
```diff
@@ -5,53 +5,74 @@ needed to build a set of RPMs that are useful for working with Geotrellis.  If
+- [`scripts`](scripts) is a directory containing scripts used to build the RPMs mentioned above, as well as the `gdal-and-friends.tar.gz` tarball.
+- [`Makefile`](Makefile) coordinates the build process.
+- The various Dockerfiles specify the various images discussed above.
+- `*.mk`: these are included in the Makefile.
```
I think this is an echo of a previous version of these docs.

CloudNiner replied:
Reverted.

CloudNiner committed:
Revert to newer files & dirs list

CloudNiner committed:
Cleanup README based on PR feedback

- Re-add specific calls to action in first
  README section
- Add SBT lighter section to provide more
  context for BootstrapAction
- Revert to the master branch version of
  "files and documents"
- Add some markdown links to aid navigation

CloudNiner committed:
Clarify the do nothing next steps

CloudNiner committed:
Satisfy markdown linter

Only one top level heading allowed

CloudNiner committed:
Re-add build+publish suggestion

CloudNiner reviewed (commented):
@jpolchlo ready for another look. The comments should mostly be addressed.

jpolchlo reviewed (approved):
Looks good
